pkgs/by-name/gr/gradia/package.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  gtk4,
+  glib,
+  ninja,
+  meson,
+  stdenv,
+  python3,
+  libportal,
+  libadwaita,
+  wrapGAppsHook4,
+  fetchFromGitHub,
+  nix-update-script,
+  blueprint-compiler,
+  desktop-file-utils,
+  gobject-introspection,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gradia";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "AlexanderVanhee";
+    repo = finalAttrs.pname;
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-9iXz4MjgG4VlystZZIKOheK4hYuz6lRXPivy+1K+cYs=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    wrapGAppsHook4
+    blueprint-compiler
+    desktop-file-utils
+    gobject-introspection
+    (python3.withPackages (
+      ps: with ps; [
+        pillow
+        pycairo
+        pygobject3
+        pygobject-stubs
+      ]
+    ))
+  ];
```

```suggestion
  nativeBuildInputs = [
        meson
    ninja
    wrapGAppsHook4
    blueprint-compiler
    desktop-file-utils
    gobject-introspection
  ];

  dependencies = [
      pillow
      pycairo
      pygobject3
      pygobject-stubs
    ];
```

if you build this as a python application, you can use the `dependencies` definition. Currently, this will almost certainly not produce a runnable executable on cross.

---

Done!